packages/react-components/react-toolbar/src/components/Toolbar/Toolbar.types.ts

```diff
@@ -48,15 +54,15 @@ export type ToolbarProps = ComponentProps<ToolbarSlots> & {
  * State used in rendering Toolbar
  */
 export type ToolbarState = ComponentState<ToolbarSlots> &
-  Required<Pick<ToolbarProps, 'size' | 'checkedValues'>> &
+  Required<Pick<ToolbarProps, 'size' | 'checkedValues' | 'vertical'>> &
   Pick<ToolbarProps, 'defaultCheckedValues' | 'onCheckedValueChange'> & {
     /*
      * Toggles the state of a ToggleButton item
      */
     handleToggleButton: ToggableHandler;
   };
 
-export type ToolbarContextValue = Pick<ToolbarProps, 'size'> & {
+export type ToolbarContextValue = Pick<ToolbarProps, 'size' | 'vertical'> & {
```

Comment:
```suggestion
export type ToolbarContextValue = Pick<ToolbarState, 'size' | 'vertical'> & {
```

When you're creating `ToolbarContextValue` you actually derive it from state, not from props. So perhaps the typings should be deriving from `ToolbarState` instead of `ToolbarProps`.

This will also ensure the types will be `Required`

Reply:
IMO state/props should derive from context values and not the other way around since it's a more primitive/simple type 🤷

Reply:
Sounds even better, as long as its not from props it makes sense to me

Reply:
`MenuContextValue` is deriving it from `MenuState` so to keep consistence I refactored here to the same: https://github.com/microsoft/fluentui/pull/24940/commits/dafd5014ee8c6d80f651acc484a572abce759c7f